src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit = require("utils/AppInit"),
+        PreferencesManager = require("preferences/PreferencesManager"),
+        HealthDataUtils = require("healthData/HealthDataUtils"),
+        uuid = require("thirdparty/uuid"),
+        UrlParams = require("utils/UrlParams").UrlParams;
+
+    var prefs = PreferencesManager.getExtensionPrefs("healthData");
+
+    prefs.definePreference("healthDataTracking", "boolean", true);
+
+    var ONE_DAY = 24 * 60 * 60 * 1000,
+        timeoutVar;
+
+    var params = new UrlParams();
+    params.parse();
+
+    /**
+     * Get the health data which will be send to the server. Initially it is only one time data.
+     */
+    function getHealthData() {
+        var result = new $.Deferred(),
+            oneTimeHealthData = {};
+
+        var guid = PreferencesManager.getViewState("GUID");
+
+        if (!guid) {
+            guid = uuid.v4();
+            PreferencesManager.setViewState("GUID", guid);
+        }
+
+        oneTimeHealthData.guid = guid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.osVersion = HealthDataUtils.getOSVersion();
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (jsonData) {
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                })
+                .always(function () {
+                    PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime());
```

Comment:
I think this should be moved into the `.always()` in `_checkHealthDataExport()`, since that's where all the other code that deals with `"lastTimeSendHealthData"` is located.